### ISSUE TITLE
Deprecate/remove Arr::sortRecursive() and array_sort_recursive()

### ISSUE THREAD

**Issue by vlakoff** [laravel/ideas#1391]
See https://github.com/laravel/framework/pull/26526, and in particular https://github.com/laravel/framework/pull/26526#issuecomment-439200067:

> Another thing, I don't like much `Arr::sortRecursive` because it sorts indexed arrays by values, and associative arrays by keys... This is quite a specific behaviour, and was designed specifically for this `assertExactJson` method.
> 
> Same goes for this new `Arr::sortAssocRecursive` method.
> 
> I think we may consider:
> 
> * deprecate/remove `Arr::sortRecursive` (and `array_sort_recursive`) as they would now be unused by Laravel,
> 
> * and move `Arr::sortAssocRecursive` to a protected method `TestResponse::sortAssocRecursive`
> 
> 
> If you agree on the principle, this could be posted on [laravel/ideas](https://github.com/laravel/ideas), or if there is a clear consensus, just directly submit a PR.

**Comment by vlakoff** [comment]
FWIW, I'll just post this alternative method, adding a sorting function parameter so you can sort as you want.

Though I don't like it much either, and should be added in user code rather than in laravel:

```php
public static function sortRecursive($array, callable $function)
{
    foreach ($array as &$value) {
        if (is_array($value)) {
            $value = static::sortRecursive($value, $function);
        }
    }

    $function($array);

    return $array;
}
```

Usage examples:
```php
$sorted = Arr::sortRecursive($data, 'sort');

$sorted = Arr::sortRecursive($data, function (&$array) {
    if (Arr::isAssoc($array)) {
        ksort($array);
    } else {
        sort($array);
    }
});
```

**Comment by HongjiangHuang** [comment]
I think this function does not have the sorting function... 
The user's callback does not necessarily do sorting. 
You function i feel more like `recursive_each `

**Comment by vlakoff** [comment]
Yeah, and it has to modify the array in-place. I tried more elaborate ways, but basically, they wouldn't support the first usage example.

As I said, I just posted this function FWIW, and wouldn't recommend adding it as a general-purpose function.
